plugins/scenario/hu.bme.mit.gamma.scenario.language/src/hu/bme/mit/gamma/scenario/language/formatting/ScenarioLanguageFormatter.xtend

```diff
@@ -0,0 +1,80 @@
+/********************************************************************************
+ * Copyright (c) 2020-2021 Contributors to the Gamma project
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ ********************************************************************************/
+package hu.bme.mit.gamma.scenario.language.formatting
+
+import com.google.inject.Inject
+import hu.bme.mit.gamma.scenario.language.services.ScenarioLanguageGrammarAccess
+import org.eclipse.xtext.formatting.impl.AbstractDeclarativeFormatter
+import org.eclipse.xtext.formatting.impl.FormattingConfig
+
+class ScenarioLanguageFormatter extends AbstractDeclarativeFormatter {
+
+	@Inject extension ScenarioLanguageGrammarAccess grammar
+
+	override protected configureFormatting(FormattingConfig config) {
+		// based on source found here: https://github.com/mn-mikke/Model-driven-Pretty-Printer-for-Xtext-Framework/wiki/Standard-Way-of-Code-Formatting-in-Xtext-Framework
+		// set a line wrap after each statechart and port assignment 
+		config.setLinewrap.after(grammar.scenarioDeclarationAccess.packageAssignment_1)
```

Comment:
It might be a good idea to introduce basic expression language formatting here as well. See the action or statechart language on how it is done.